packages/snage/src/create/consoleParamsReader.ts

```diff
@@ -90,21 +90,25 @@ const handleMissingValue = async (field: Field, consoleArguments: {}, returnValu
             returnValues[field.name] = fieldValue;
         }
     } else {
-        returnValues[field.name] = '';
+        returnValues[field.name] = null;
     }
     return right(true);
 };
 
-const getAdjustedMetaData = (metaValues: Record<string, unknown>, fields: Field[]): Record<string, unknown> => {
-    const adjustedRecords: Record<string, unknown> = {};
+const getAdjustedData = (metaValues: Record<string, unknown>, fields: Field[]):FieldForOutput[]  => {
+    const adjustedRecords: FieldForOutput[] = [];
     fields
         .filter((field) => field.name in metaValues)
-        .map((field) => {
-            if (field.optional && metaValues[field.name] == '') {
-                adjustedRecords['# ' + field.name] = null;
-            } else {
-                adjustedRecords[field.name] = metaValues[field.name];
-            }
+        .forEach((field) => {
+            adjustedRecords.push({optional: !!field.optional, list: !!field.list, name: field.name, value: metaValues[field.name]});
         });
     return adjustedRecords;
 };
+
+//FIXME I'm not happy with that name since it bleeds the purpose for later down the line instead of accurately describing it. Any suggestions?
+export interface FieldForOutput {
```

Comment:
Wieso braucht man das, könnte man nicht fieldvalues und felder in 2 parametern übergeben?